### DISCUSSION TITLE
Applying post processing on acquired images

### DISCUSSION THREAD

**Discussion by jacopoabramo** [ImSwitch/ImSwitch#107] [Q&A]
Greetings,

As the title mentioned I had in my mind the idea of applying a post-processing function to the acquired images from the cameras. One example would be to perform a frame subtraction to apply a median filter on the images. I've seen somewhere in ImSwitch that maybe this is possible (i.e. define your post-processing function and apply it while doing the liveview/recording), but I don't know exactly how I can implement this or if I got the wrong idea.

Thanks!

**Comment by xcasas** [comment]
Hey @jacopoabramo,

I don't know if this is the best solution, but I thought one way to do it could be in the liveview thread of the camera. If you check in [DetectorsManager](https://github.com/kasasxav/ImSwitch/blob/master/imswitch/imcontrol/model/managers/DetectorsManager.py), around line 155. That executes the function `updateLatestFrame` in the detectors. I was thinking, if you add in your camera a detector parameter for example, saying postprocessing and with options Off, median or subtraction. Then, from the function `updateLatestFrame `in your `DetectorManager` add if statements and perform the processing. And then a similar option if you want to have it for the recording in `getChunk`. What do you think?

**Reply by jacopoabramo** [reply]
I'll try this out once I got some time, see how it works!

**Comment by jacopoabramo** [comment]
I'm updating this discussion as I managed to use the ImSwitch Pyro5 server to send requests from the XimeaManager in order to create a sequence of movements on the X-Y axis, creating a median of a stack of images and applying them directly to the acquired images. You can check the code [here](https://github.com/Eggeling-Lab-Microscope-Software/ImSwitch/blob/master/imswitch/imcontrol/model/managers/detectors/XimeaManager.py).
